x-pack/plugins/cases/public/components/category/category_component.tsx

```diff
@@ -26,15 +29,11 @@ export const CategoryComponent: React.FC<CategoryComponentProps> = React.memo(
       }));
     }, [availableCategories]);
 
-    const [selectedOptions, setSelectedOptions] = useState<Array<EuiComboBoxOptionOption<string>>>(
-      category != null ? [{ label: category }] : []
-    );
+    const selectedOptions = category != null ? [{ label: category }] : [];
 
     const onComboChange = useCallback(
       (currentOptions: Array<EuiComboBoxOptionOption<string>>) => {
         const value = currentOptions[0]?.label;
-
-        setSelectedOptions(currentOptions);
```

Comment:
This breaks the tests in `category_component.test.tsx` because the component is not rendered and we don't see the value 😅 

What was the reasoning for this change, precisely to avoid a rerender?

I can update the tests to assert only that `onChange` gets called with the correct values.

Reply:
Yes, I made the component a pure component so the `CategoryFormField` component can control the state using the `field.setValue` provided by the form library.

>I can update the tests to assert only that onChange gets called with the correct values.

Sounds good to me.